packages/app/ui/components/BigInput.tsx

```diff
@@ -149,9 +151,26 @@ export function BigInput({
         metadata.title = title;
       }
 
-      // Always include image field for notebooks, even if null
-      // This ensures we can clear an image by setting it to null
-      metadata.image = imageUri;
+      if (imageUri) {
+        const attachment = attachments.find(
+          (attachment) =>
+            attachment.type === 'image' &&
+            attachment.uploadState?.status === 'success' &&
+            attachment.file?.uri === imageUri
+        );
+        if (
+          attachment &&
+          attachment.type === 'image' &&
+          attachment.uploadState &&
+          'remoteUri' in attachment.uploadState
+        ) {
+          metadata.image = attachment.uploadState.remoteUri;
+        }
+      } else {
+        // Always include image field for notebooks, even if null
+        // This ensures we can clear an image by setting it to null
+        metadata.image = imageUri;
```

Comment:
if this branch only gets hit when `imageUri` is `null`, can we set it to that explicitly to avoid confusion?

Reply:
yeah, good point.